### ISSUE TITLE
[Feature]: Publish Docker ARM images as multi-architecture images

### ISSUE THREAD

**Issue by deviantintegral** [microsoft/playwright#29819]
### 🚀 Feature Request

Currently, [Docker images for Playwright](https://mcr.microsoft.com/en-us/product/playwright/tags) are split by architecture.

<img width="1365" alt="image" src="https://github.com/microsoft/playwright/assets/255023/cd1cc846-03c2-4de7-917a-266ea1d3e4e2">

However, this complicates sharing Docker and Docker Compose files with others, especially macOS users. Before starting, they have to manually modify any `FROM` or `image:` lines to update the architecture.

Docker has this problems solved though. As long as a [Multi-arch image is built](https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/), clients will choose the appropriate architecture automatically unless it's overridden.

### Example

`docker pull mcr.microsoft.com/playwright:v1.42.1` should pull the right native architecture automatically.

### Motivation

This will reduce confusion for Playwright users, and may even reduce bug reports. For example, it's not unheard of for there to be errors emulating amd64 software on QEMU or Rosetta. It will also improve performance for users who are successfully running Playwright under emulation.

**Comment by mxschmitt** [comment]
We publish Docker images with a manifest already, so the `v1.42.1` Docker tag links to `v1.42.1-arm64` and `v1.42.1-amd64`. Based on your arch, it will pull the right one.

If I do `docker run -it mcr.microsoft.com/playwright:v1.42.1 /bin/bash -c "arch"` it prints a aarch64 for me.

Also `docker image inspect mcr.microsoft.com/playwright:v1.42.1` does. Could you give us repro steps?

**Comment by deviantintegral** [comment]
Yes, I get the right architecture too. Sorry about that - when I filed, I was on mobile data and was going off of what is visible on the MCR site.

Looking further, I actually think this is a design issue with how MCR presents images.

<img width="1456" alt="image" src="https://github.com/microsoft/playwright/assets/255023/48a96e05-81d1-4bd7-949d-df220869094d">

As a comparison, here's Docker Hub. It's easy to see what architectures `latest` points to:

<img width="868" alt="image" src="https://github.com/microsoft/playwright/assets/255023/1690e5b8-3daa-4a96-98ba-a5926d0e308e">

Or here's GitHub's registry, which shows the manifest list at the top and the associated images in a table:

https://github.com/Lissy93/web-check/pkgs/container/web-check/187292771?tag=latest

<img width="1063" alt="image" src="https://github.com/microsoft/playwright/assets/255023/fe60749f-79e7-4f0e-b93d-23fdcbe9fdae">

What's missing from the MCR listing is a connection between the manifest and the images it contains.

I'm also guessing this means you are required to publish Docker tags for each architecture? Because if all I saw as a user was tags with no architecture, I'd assume by default they were multiarch images.

**Comment by mxschmitt** [comment]
Yeah these manifests could be clearer, I recommend filing an issue here: https://github.com/microsoft/containerregistry/issues

We as Playwright publish our image on both architectures. 

Looks like there are no action items from our side here.

**Comment by deviantintegral** [comment]
To use Ubuntu as an example, none of their docker tags have an architecture included in them. If you want to pull a specific architecture, you would run something like `docker pull --platform linux/amd64 ubuntu:latest`. Could Playwright do the same?

**Comment by mxschmitt** [comment]
docker pull --platform linux/amd64 mcr.microsoft.com/playwright:v1.42.1-jammy

This will work with Playwright as well. Thats actually our recommendation. The customers should not need to worry about the arm64/amd64 images, these should be more hidden - hence we never referenced to them.

For some reason Docker Hub is not visualising the intermediate Docker images. Not sure what kind of magic they do, since MCR (Microsoft Container Registry) - what we use - and DockerHub are significant different, this might be due to this difference.

**Comment by rfay** [comment]
On an Apple Silicon machine ...

> docker pull --platform linux/amd64 mcr.microsoft.com/playwright:v1.42.1-jammy

That actually only works on Docker providers that will do Rosetta. And it's typically not very reliable. Very strange things can happen.

I encourage you to make multi-arch images available. It's done with `docker buildx build --push` and well documented for hub.docker.com anyway, pushing multiple manifests to a single tag.

This has been happening now for four years... https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/

**Comment by mxschmitt** [comment]
@rfay We already publish multi-arch Docker images. You can use them without specifying the platform and it will pull the correct one. e.g.

    docker pull mcr.microsoft.com/playwright:v1.42.1-jammy

will download `linux/arm64` on my macOS Silicon machine.

**Comment by rfay** [comment]
So it does, thanks, sorry for not understanding. Your web UI doesn't imply that at all, and the fact that you call out arm64 images with separate names:

<img width="1142" alt="image" src="https://github.com/microsoft/playwright/assets/112444/95eb95d6-b7db-425e-ab16-7efbc59177dd">

**Comment by dgozman** [comment]
Closing as per above.

**Comment by deviantintegral** [comment]
Agreed! And I filed https://github.com/microsoft/containerregistry/issues/159 upstream for this. Thanks for all the feedback.
